### ISSUE TITLE
Unsatisfiable Indentation module scenario (Checkstyle 5.8 or 5.9 regression)

### ISSUE THREAD

**Issue by asheldon** [checkstyle/checkstyle#321]
There seems to be an issue with indentation tracking when multiple lines are used as part of annotations.  Here's an example with Jackson annotations:

    import com.fasterxml.jackson.annotation.JsonSubTypes;
    import com.fasterxml.jackson.annotation.JsonSubTypes.Type;

    @JsonSubTypes({
        @Type(value = Boolean.class, name = "boolean"),
        @Type(value = String.class, name = "string") })
    public class CheckstyleIndentationError {
    }

then checkstyle incorrectly says 
> "'public' have incorrect indentation level 0, expected level should be 4."

    import com.fasterxml.jackson.annotation.JsonSubTypes;
    import com.fasterxml.jackson.annotation.JsonSubTypes.Type;

    @JsonSubTypes({
        @Type(value = Boolean.class, name = "boolean"),
        @Type(value = String.class, name = "string") })
        public class CheckstyleIndentationError {
    }

After indenting 'public' by 4 spaces, checkstyle correctly says 
> "'class def modifier' have incorrect indentation level 4, expected level should be 0." 

This does not reproduce in Checkstyle 5.7, not tested on Checkstyle 5.8. 

    <module name="Checker">
        <module name="TreeWalker">
            <module name="Indentation"/>
        </module>
    </module>

**Comment by tole42** [comment]
Is the following annotation still an issue with 6.2/6.3 , or is my indentation wrong?

```java
@MessageDriven(
    mappedName = "java:jboss/exported/jms/topic/mytopic",
    activationConfig =
        {
            @ActivationConfigProperty(propertyName="destinationType", propertyValue="javax.jms.Topic"),
            @ActivationConfigProperty(propertyName="destinationLookup", propertyValue= JmsService.JMS_TOPIC_MYTOPIC),
            @ActivationConfigProperty(propertyName = "acknowledgeMode", propertyValue = "Auto-acknowledge")
        }
)
public class FooMessageDrivenBean implements MessageListener {
```

**Comment by romani** [comment]
@tole42 , please post configuration and checkstyle output. We need exact test case(config+input) to review that problem.

**Comment by tole42** [comment]
Output:
```
 (indentation) Indentation: ')' have incorrect indentation level 0, expected level should be 4.
```
Line number is the line before `public class FooMessageDrivenBean` [The line with the ")"].
Checkstyle configuration: https://gist.github.com/tole42/b34e3efea05cf482e8e9

**Comment by mkordas** [comment]
The root cause is probably the same as in #553. @romani, do we need both issues open? I think we can merge them.

**Comment by romani** [comment]
thanks a lot. lets do all updates in #533
